src/xrGame/DiscordRPC.cpp

```diff
@@ -0,0 +1,50 @@
+#include "stdafx.h"
+
+#include "DiscordRPC.h"
+#include "xrCore/Text/StringConversion.hpp"
+
+#include <DiscordRPC/discord_register.h>
+#include <DiscordRPC/discord_rpc.h>
+
+constexpr pcstr DISCORD_APP_ID = "869824972464488510";
+
+DiscordRPC g_DiscordRPC;
+
+void DiscordRPC::Init()
+{
+    xr_sprintf(build_name, "build %u", Core.GetBuildId());
+    start_time = time(nullptr);
+    DiscordEventHandlers handle{};
+    Discord_Initialize(DISCORD_APP_ID, &handle, 1, nullptr);
+}
+
+void DiscordRPC::Deinit()
+{
+    Discord_ClearPresence();
+    Discord_Shutdown();
+}
+
+void DiscordRPC::Update(bool type, pcstr data)
+{
+    DiscordRichPresence discordPresence{};
+    static std::locale locale("");
+    xr_string temp = StringToUTF8(data, locale);
+
+    discordPresence.startTimestamp  = start_time;
+    discordPresence.smallImageText  = build_name;
+    discordPresence.largeImageKey   = "current_level_name";
+    discordPresence.smallImageKey   = "build_name";
+
+    //  type = TRUE  -> setting location
+    //  type = FALSE -> setting task
+    if (type)
+        xr_strcpy(current_level_name, temp.c_str());
+    else
+        xr_strcpy(current_task_name, temp.c_str());
```

Comment:
Looks like it can be merged into conditional operator, like this:
`xr_strcpy(type ? current_level_name : current_task_name, temp.c_str());`

But I'd leave @Xottab-DUTY to decide If we need this adjustment.

Reply:
Using ternary operator is no longer relevant, since now we have enumeration instead of bool :)